### ISSUE TITLE
Using gpu is slower as not using it

### ISSUE THREAD

**Issue by sonic182** [rustformers/llm#357]
I have build the model from source, first without the --features cublas flag, then I rebuild with it.

```
$ llm infer --use-gpu -a mpt -m models/mpt-7b-instruct-q5_1.bin -p "Rust is a cool programming language because" -t 6 --stats
⣾ Loaded hyperparametersggml_init_cublas: found 1 CUDA devices:
  Device 0: NVIDIA GeForce RTX 2060
✓ Loaded 194 tensors (5.0 GB) after 1697ms
Rust is a cool programming language because you can build real-time systems with it. It's the only general...
```

maybe is a cache stuff? where is the cache of the models after they're run the first time?

**Comment by LLukas22** [comment]
See https://github.com/rustformers/llm/pull/325. Cublas/Clblast acceleration isn't currently supported on the main branch, meaning you can build with the acceleration enabled but it wont accelerate the inference. Also only `llama` based models are currently completely accelerated as `ggml` doesn't support gpu acceleration for certain matrix operations needed for other model architectures.

**Comment by sonic182** [comment]
Ok thanks!!

Maybe is interesting to print some information about it, when it is a non llama model, to print "this model cannot be gpu accelerated" or something like that, with llm-cli

**Comment by LLukas22** [comment]
Yeah i was planning to create a table in the "accelerators docu" which shows which architecture can be accelerated by which gpu backend, as it's likely that some models will get support for e.g. cuda earlier than clblast.

**Comment by KerfuffleV2** [comment]
In addition to the previous points, if you have a good CPU and a relatively weak GPU then offloading to the GPU can still be slower even if a model has full GPU support.
